### PR TITLE
Use PyInstaller to build client distributions

### PR DESCRIPTION
**Pull request by toidi**
To build distribution, install PyInstaller(`python -m pip install pyinstaller`) and run command:

    python -m PyInstaller tx.spec

`dist\tx.exe` executable will be created as the result of build process.

### CODE REVIEW

**Comment by toidi** [comment]
Sample build log

    Microsoft Windows [Version 6.3.9600]
    (c) Корпорация Майкрософт (Microsoft Corporation), 2013. Все права защищены.
    
    C:\Users\Ksenia>cd AppData\Local\Programs\Python\Python35-32
    
    C:\Users\Ksenia\AppData\Local\Programs\Python\Python35-32>dir
     Том в устройстве C не имеет метки.
     Серийный номер тома: 408D-B857
    
     Содержимое папки C:\Users\Ksenia\AppData\Local\Programs\Python\Python35-32
    
    01.11.2015  22:00    <DIR>          .
    01.11.2015  22:00    <DIR>          ..
    01.11.2015  21:59    <DIR>          DLLs
    01.11.2015  21:59    <DIR>          Doc
    01.11.2015  21:58    <DIR>          include
    01.11.2015  21:59    <DIR>          Lib
    01.11.2015  21:58    <DIR>          libs
    13.09.2015  02:20            30 043 LICENSE.txt
    13.09.2015  01:06           282 293 NEWS.txt
    13.09.2015  02:17            38 168 python.exe
    13.09.2015  02:17            51 480 python3.dll
    13.09.2015  02:17         3 128 600 python35.dll
    13.09.2015  02:17            38 168 pythonw.exe
    13.09.2015  01:06             6 740 README.txt
    01.11.2015  22:00    <DIR>          Scripts
    01.11.2015  21:59    <DIR>          tcl
    01.11.2015  21:59    <DIR>          Tools
    25.06.2015  23:34            85 328 vcruntime140.dll
                   8 файлов      3 660 820 байт
                  10 папок  108 208 164 864 байт свободно
    
    C:\Users\Ksenia\AppData\Local\Programs\Python\Python35-32>python
    Python 3.5.0 (v3.5.0:374f501f4567, Sep 13 2015, 02:16:59) [MSC v.1900 32 bit (Intel)] on win32
    Type "help", "copyright", "credits" or "license" for more information.
    >>> exit()
    
    C:\Users\Ksenia\AppData\Local\Programs\Python\Python35-32>python -m pip install pyinstaller
    Collecting pyinstaller
      Downloading PyInstaller-3.0.tar.gz (2.7MB)
        100% |################################| 2.7MB 154kB/s
    Requirement already satisfied (use --upgrade to upgrade): setuptools in c:\users\ksenia\appdata\local\programs\python\py
    thon35-32\lib\site-packages (from pyinstaller)
    Collecting pypiwin32 (from pyinstaller)
      Downloading pypiwin32-219-cp35-none-win32.whl (7.9MB)
        100% |################################| 7.9MB 50kB/s
    Installing collected packages: pypiwin32, pyinstaller
      Running setup.py install for pyinstaller
    Successfully installed pyinstaller-3.0 pypiwin32-219
    
    C:\Users\Ksenia\Downloads>cd transifex-client-tx-pyinstaller
    
    C:\Users\Ksenia\Downloads\transifex-client-tx-pyinstaller>cd transifex-client-tx-pyinstaller
    
    C:\Users\Ksenia\Downloads\transifex-client-tx-pyinstaller\transifex-client-tx-pyinstaller>C:\Users\Ksenia\AppData\Local\Programs\Python\Python35-32\python -m PyInstaller tx.spec
    234 INFO: PyInstaller: 3.0
    234 INFO: Python: 3.5.0
    234 INFO: Platform: Windows-post2012Server-6.3.9600
    234 INFO: UPX is not available.
    234 INFO: Extending PYTHONPATH with C:\Users\Ksenia\Downloads\transifex-client-tx-pyinstaller\transifex-client-tx-pyinst
    aller
    249 INFO: checking Analysis
    249 INFO: Building Analysis because out00-Analysis.toc is non existent
    249 INFO: Initializing module dependency graph...
    249 INFO: Initializing module graph hooks...
    249 INFO: Analyzing base_library.zip ...
    3606 INFO: running Analysis out00-Analysis.toc
    4856 INFO: Analyzing tx
    5275 INFO: Processing pre-find module path hook   distutils
    6072 INFO: Looking for import hooks ...
    6088 INFO: Processing hook   hook-encodings.py
    6103 INFO: Processing hook   hook-pydoc.py
    6103 INFO: Processing hook   hook-xml.py
    6353 INFO: Processing hook   hook-distutils.py
    6353 INFO: Processing hook   hook-sysconfig.py
    6353 INFO: Processing hook   hook-xml.sax.py
    6400 INFO: Looking for ctypes DLLs
    6416 INFO: Analyzing run-time hooks ...
    6431 INFO: Including run-time hook 'pyi_rth_pkgres.py'
    6447 INFO: Looking for dynamic libraries
    8430 INFO: Looking for eggs
    8430 INFO: Using Python library C:\Users\Ksenia\AppData\Local\Programs\Python\Python35-32\python35.dll
    8430 INFO: Found binding redirects:
    []
    8461 INFO: Warnings written to C:\Users\Ksenia\Downloads\transifex-client-tx-pyinstaller\transifex-client-tx-pyinstaller
    \build\tx\warntx.txt
    8945 INFO: Appending 'datas' from .spec
    8992 INFO: checking PYZ
    8992 INFO: Building PYZ because out00-PYZ.toc is non existent
    8992 INFO: Building PYZ (ZlibArchive) C:\Users\Ksenia\Downloads\transifex-client-tx-pyinstaller\transifex-client-tx-pyin
    staller\build\tx\out00-PYZ.pyz
    9320 INFO: checking PKG
    9320 INFO: Building PKG because out00-PKG.toc is non existent
    9320 INFO: Building PKG (CArchive) out00-PKG.pkg
    12102 INFO: Updating manifest in C:\Users\Ksenia\AppData\Roaming\pyinstaller\bincache00_py35_32bit\python35.dll
    12117 INFO: Updating resource type 24 name 2 language 1033
    15195 INFO: Bootloader C:\Users\Ksenia\AppData\Local\Programs\Python\Python35-32\lib\site-packages\PyInstaller\bootloade
    r\Windows-32bit\run.exe
    15195 INFO: checking EXE
    15195 INFO: Building EXE because out00-EXE.toc is non existent
    15195 INFO: Building EXE from out00-EXE.toc
    15367 INFO: Appending archive to EXE C:\Users\Ksenia\Downloads\transifex-client-tx-pyinstaller\transifex-client-tx-pyins
    taller\dist\tx.exe
    
    C:\Users\Ksenia\Downloads\transifex-client-tx-pyinstaller\transifex-client-tx-pyinstaller>cd dist
    
    C:\Users\Ksenia\Downloads\transifex-client-tx-pyinstaller\transifex-client-tx-pyinstaller\dist>dir
     Том в устройстве C не имеет метки.
     Серийный номер тома: 408D-B857
    
     Содержимое папки C:\Users\Ksenia\Downloads\transifex-client-tx-pyinstaller\transifex-client-tx-pyinstaller\dist
    
    01.11.2015  22:15    <DIR>          .
    01.11.2015  22:15    <DIR>          ..
    01.11.2015  22:15         6 370 315 tx.exe
                   1 файлов      6 370 315 байт
                   2 папок  108 137 660 416 байт свободно
    
    C:\Users\Ksenia\Downloads\transifex-client-tx-pyinstaller\transifex-client-tx-pyinstaller\dist>tx.exe --help
    Usage: tx.exe [options] command [cmd_options]
    
    This is the Transifex command line client which allows you to manage your
    translations locally and sync them with the master Transifex server. If you'd
    like to check the available commands issue `tx.exe help` or if you just want
    help with a specific command issue `tx.exe help command`
    
    Options:
      --version             show program's version number and exit
      -h, --help            show this help message and exit
      -d, --debug           enable debug messages
      -q, --quiet           don't print status messages to stdout
      -r ROOT_DIR, --root=ROOT_DIR
                            change root directory (default is cwd)
      --traceback           print full traceback on exceptions
      --disable-colors      disable colors in the output of commands
    
    C:\Users\Ksenia\Downloads\transifex-client-tx-pyinstaller\transifex-client-tx-pyinstaller\dist>

**Comment by toidi** [comment]
1. Download transifex-client sources via git or github archive.

  a. `git clone https://github.com/transifex/transifex-client.git`
  b. Download and unpack https://github.com/transifex/transifex-client/archive/master.zip

2. Download and install [python](https://www.python.org/downloads/windows/).

 At this step choose apropriate version of python: 2 or 3 and x86 or x86-64 instruction set.

 Make sure pip marked for installation(python 3 only).

3. Install PyInstaller (python3 tested).

 Suppose that python installed to `C:\Users\Ksenia\AppData\Local\Programs\Python\Python35-32`

 Make `python.exe` accessible via PATH environment variable or cd to directory containing python.exe.

  ~~~
  python -m pip install pyinstaller
  ~~~

 This command will install `PyInstaller` package and its dependencies.

4. Build transifex-client distribution.

  Change directory to transifex-client folder and run command:
  ~~~
  python -m PyInstaller tx.spec
  ~~~

**Comment by diegobz** [comment]
@toidi Good stuff. I could generate a working tx.exe. About the instructions above, could you add it to the read me file on its own section, instead?

**Comment by toidi** [comment]
@diegobz, updated commit with readme and icon file.

Also rebased ci branch.

File: README.rst [hunk 1]
```diff
@@ -26,4 +26,46 @@ You can also install the `in-development version`_ of transifex-client
...
+
+   This command will install ``PyInstaller`` package and its dependencies.
+
+4. Build ``transifex-client`` distribution.
+
+   Change directory to transifex-client folder and run command:
+
+   ::
+
+     python -m PyInstaller tx.spec
```
Comment (diegobz) [inline README.rst]: `pyinstaller contrib/tx.spec` as tx.spec is now under contrib.

**Comment by diegobz** [comment]
`setup.py` still has reference to `py2exe`. Should this all be removed?

File: contrib/tx_test.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+HOST="https://www.transifex.com"
```
Comment (diegobz) [inline contrib/tx_test.sh]: Rename to `test_win_build.sh` and move it under `contrib`.

Comment (toidi) [inline contrib/tx_test.sh]: Yes. I removed unused code.

File: README.rst (hunk 1 above)
Comment (toidi) [inline README.rst]: fixed

**Pull request merged**